### PR TITLE
Automatic detection of annotation without checkpoints

### PR DESCRIPTION
**Pull request by Maarten-vd-Sande**
**What problem is the PR solving / What's new?**
Now we should be able to automatically detect whether or not an assembly has an annotation :tada:, no stupid if statement required + other pipelines' trackhub now can also get annotations :).

**What did change**
Function that queries genomepy for an annotation of an assembly. Should the function crash or return false when no genome has been found?

### CODE REVIEW

File: seq2science/rules/get_genome.smk [hunk 1]
```diff
@@ -118,3 +127,31 @@ rule decoy_transcripts:
...
+def has_annotation(assembly):
+    """
+    returns True/False on whether or not the assembly has an annotation.
+    """
+    # check if genome providd by user or already downloaded, if so check if the annotation came along
+    if all(os.path.exists(f"{config['genome_dir']}/{assembly}.{extension}") for extension in config["genome_types"]):
+        return os.path.exists(f"{config['genome_dir']}/{assembly}.annotation.gtf")
+
+    if "provider" in config:
+        providers = config["provider"]
```
Comment (siebrenf) [inline seq2science/rules/get_genome.smk:142]: should this not be a list?

`providers = [config["provider"]]`

Comment (Maarten-vd-Sande) [reply seq2science/rules/get_genome.smk]: correct

File: seq2science/rules/get_genome.smk [hunk 2]
```diff
@@ -118,3 +127,31 @@ rule decoy_transcripts:
...
+        providers = ["Ensembl", "UCSC", "NCBI"]
+
+    # check if we expect an annotation
+    # we do not want genomepy outputting to us that it is downloading stuff
+    with open(os.devnull, 'w') as null:
+        with contextlib.redirect_stdout(null), contextlib.redirect_stderr(null):
+            for provider in providers:
+                p = genomepy.ProviderBase.create(provider)
+                if assembly in p.genomes and \
+                    p.get_genome_download_link(assembly) is not None:
```
Comment (siebrenf) [inline seq2science/rules/get_genome.smk:153]: isn't this double? If the assembly is in p.genomes, there should always be a genome_download_link.

Comment (Maarten-vd-Sande) [reply seq2science/rules/get_genome.smk:153]: I guess? I don't know how genomepy works internally :). I had some issues with just using get_genome_download_link, and added the check if its in genomes. I didn't make the effort to check if it's superfluous. :angel:

**Comment by Maarten-vd-Sande** [comment]
Tnx for thorough feedback :)

**Comment by Maarten-vd-Sande** [comment]
So you mean to log a warning if no annotation is found? That makes sense, I will add it.

**Pull request merged**